src/lib.rs

```diff
@@ -324,17 +322,14 @@ impl Keccak {
 
     // squeeze output
     pub fn squeeze(&mut self, output: &mut [u8]) {
-        let outlen = output.len();
-        let rate = self.rate;
-
         // second foldp
         let mut op = 0;
-        let mut l = outlen;
-        while l >= rate {
-            setout(self.a_bytes(), &mut output[op..], rate);
+        let mut l = output.len();
+        while l >= self.rate {
```

Comment:
Just out of curiosity: did this change also improve anything? Kind of liked it better with less repetitions of `self.`